File: src/app/models/remote.ts
import { User } from "./user";

export interface Remote {
  id: number,
     user : User,
     fromDatetime : Date,
     toDatetime : Date,
     date: string,
     type : String,
     reason : string,
     totalHours : number,
     evident : string,
     approver_id: string,
     approver: string,
     comment: string,
      status: string,
      workedHours: number,
      createAt : Date,
  }
  

File: sys/src/main/java/com/hrms/sys/models/Remote.java
package com.hrms.sys.models;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "remotes")
@Data
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Remote extends BaseEntity{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @Column(name = "from_datetime")
    private LocalDateTime fromDatetime;

    @Column(name = "to_datetime")
    private LocalDateTime toDatetime;

    @Column(name = "allowed_hours")
    private Float allowedHours;

    @Column(name = "worked_hours")
    private Float workedHours;

    private String reason;

    private String evident;

    private String type;



    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "approver_id", referencedColumnName = "id")
    private User approver;

    private String comment;

    private String status;

    @Column(name = "code")
    private String code;






}
